src/services/codefixes/fixAddVoidToPromise.ts

```diff
@@ -0,0 +1,82 @@
+/* @internal */
+namespace ts.codefix {
+    const fixName = "addVoidToPromise";
+    const fixId = "addVoidToPromise";
+    const errorCodes = [
+        Diagnostics.Expected_0_arguments_but_got_1_Did_you_forget_to_include_void_in_your_type_argument_to_Promise.code
+    ];
+    registerCodeFix({
+        errorCodes,
+        fixIds: [fixId],
+        getCodeActions(context) {
+            const changes = textChanges.ChangeTracker.with(context, t => makeChange(t, context.sourceFile, context.span, context.program));
+            if (changes.length > 0) {
+                return [createCodeFixAction(fixName, changes, Diagnostics.Add_void_to_Promise_resolved_without_a_value, fixId, Diagnostics.Add_void_to_all_Promises_resolved_without_a_value)];
+            }
+        },
+        getAllCodeActions(context: CodeFixAllContext) {
+            return codeFixAll(context, errorCodes, (changes, diag) => makeChange(changes, diag.file, diag, context.program, new Set()));
+        }
+    });
+
+    function makeChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, span: TextSpan, program: Program, seen?: Set<ParameterDeclaration>) {
+        const node = getTokenAtPosition(sourceFile, span.start);
+        if (!isIdentifier(node) || !isCallExpression(node.parent) || node.parent.expression !== node || node.parent.arguments.length !== 0) return;

```

Comment:
Wasn't this checked when the diagnostic was produced?

Reply:
Yes, but I'd rather recheck then blindly assert. The the place where we produce this error is so distant from this location, it would be easy for someone to change something and break this. Most of the checks are fairly trivial though.